From be0e2cbc81b089104ad4caceace00e6a61e8f7de Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:04:00 -0400
Subject: [fs] btrfs: cut down stack usage in btree_write_cache_pages

Message-id: <1407179109-13714-354-git-send-email-sandeen@redhat.com>
Patchwork-id: 87244
O-Subject: [RHEL7.1 PATCH 353/422] Btrfs: cut down stack usage in btree_write_cache_pages
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Chris Mason <clm@fb.com>

commit 0e378df15cd87f540f1ba9503e4aa039e1c72741
Author: Chris Mason <clm@fb.com>
Date:   Mon May 19 20:55:27 2014 -0700

    Btrfs: cut down stack usage in btree_write_cache_pages

    This adds noinline_for_stack to two helpers used by
    btree_write_cache_pages.  It shaves us down from 424 bytes on the
    stack to 280.

    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/extent_io.c b/fs/btrfs/extent_io.c
index 7bcf6a8..62fede1 100644
--- a/fs/btrfs/extent_io.c
+++ b/fs/btrfs/extent_io.c
@@ -3399,9 +3399,10 @@ void wait_on_extent_buffer_writeback(struct extent_buffer *eb)
       TASK_UNINTERRUPTIBLE);
 }
 
-static int lock_extent_buffer_for_io(struct extent_buffer *eb,
-         struct btrfs_fs_info *fs_info,
-         struct extent_page_data *epd)
+static noinline_for_stack int
+lock_extent_buffer_for_io(struct extent_buffer *eb,
+     struct btrfs_fs_info *fs_info,
+     struct extent_page_data *epd)
 {
  unsigned long i, num_pages;
  int flush = 0;
@@ -3509,7 +3510,7 @@ static void end_bio_extent_buffer_writepage(struct bio *bio, int err)
 
 }
 
-static int write_one_eb(struct extent_buffer *eb,
+static noinline_for_stack int write_one_eb(struct extent_buffer *eb,
    struct btrfs_fs_info *fs_info,
    struct writeback_control *wbc,
    struct extent_page_data *epd)
-- 
1.7.1